### PR TITLE
ext/openssl: documentation

### PR DESCRIPTION
**Pull request by vbatts**
age old rebase

ping @zzak 
:heart:

### CODE REVIEW

File: ext/openssl/ossl_asn1.c [hunk 1]
```diff
@@ -1382,13 +1382,13 @@ ossl_asn1obj_s_register(VALUE self, VALUE oid, VALUE sn, VALUE ln)
     return Qtrue;
 }
 
-/* Document-method: OpenSSL::ASN1::ObjectId#sn
+/* Document-method: OpenSSL::ASN1::ObjectId.sn
```
Comment (nobu) [inline ext/openssl/ossl_asn1.c]: Wrong, this is not a class method.

File: ext/openssl/ossl_asn1.c [hunk 2]
```diff
@@ -1403,13 +1403,13 @@ ossl_asn1obj_get_sn(VALUE self)
     return ret;
 }
 
-/* Document-method: OpenSSL::ASN1::ObjectId#ln
+/* Document-method: OpenSSL::ASN1::ObjectId.ln
```
Comment (nobu) [inline ext/openssl/ossl_asn1.c]: Same as #sn.

File: ext/openssl/ossl_asn1.c [hunk 3]
```diff
@@ -1808,6 +1808,10 @@ Init_ossl_asn1(void)
      * the final octet of the +BitString+'s +value+.
      *
      * == OpenSSL::ASN1::ObjectId
+     * 
+     * NOTE: While OpenSSL::ASN1::ObjectId.new will allocate a new ObjectId,
+     * it is not typically allocated this way, but rather that are recieved from
```
Comment (nobu) [inline ext/openssl/ossl_asn1.c]: These lines have trailing spaces.

File: ext/openssl/ossl_bn.c [hunk 4]
```diff
@@ -121,13 +128,15 @@ ossl_bn_alloc(VALUE klass)
...
 
-/*
- * call-seq:
- *    BN.new => aBN
- *    BN.new(bn) => aBN
- *    BN.new(integer) => aBN
- *    BN.new(string) => aBN
- *    BN.new(string, 0 | 2 | 10 | 16) => aBN
+/* Document-method: OpenSSL::BN.new
+ *
```
Comment (nobu) [inline ext/openssl/ossl_bn.c]: This line has a trailing space.

File: ext/openssl/ossl_bn.c [hunk 5]
```diff
@@ -419,7 +428,9 @@ BIGNUM_2c(mod_inverse)
 
 /*
  * call-seq:
- *    bn1 / bn2 => [result, remainder]
+ *    bn1 / bn2   -> [result, remainder]
```
Comment (nobu) [inline ext/openssl/ossl_bn.c]: Does `->` differ than `=>`?

**Comment by vbatts** [comment]
@nobu thanks. I have updated. PTAL

File: ext/openssl/ossl_bn.c (hunk 5 above)
Comment (drbrain) [inline ext/openssl/ossl_bn.c]: No, rdoc recognizes both as an arrow

**Pull request closed without merging**